src/harness/simple_crl_server.py

```diff
@@ -0,0 +1,308 @@
+
+  revoke_cert_command = "cd {0} && ./revoke_and_generate_crl.sh -u".format(
+      getCertsDirectoryAbsolutePath())
+  command_exit_status = subprocess.call(revoke_cert_command, shell=True)
+  if command_exit_status:
+    raise Exception("RunTimeError:Generate CRL Chain is failed:exit_code:{}".format(
+        command_exit_status))
+
+def getCertNameToBlacklist():
+  """ Allows the user to select the certificate to blacklist from a displayed list. """
+
+  # Gets the all certificate files that are ends with .cert present in the default certs directory.
+  cert_files = [cert_file_name for cert_file_name in os.listdir(getCertsDirectoryAbsolutePath())
+                if cert_file_name.endswith('.cert')]
+
+  # Creates the mapping menu ID for each certificate file to displayed list.
+  cert_files_with_options = dict(zip(range(len(cert_files)), sorted(cert_files)))
+  print "select the certificate to blacklist:"
+  print "\n".join("[%s] %s" % (cert_file_option_id, cert_file_name)
+                  for (cert_file_option_id, cert_file_name) in
+                  sorted(cert_files_with_options.iteritems()))
+  option_id = int(raw_input(CLI_PROMPT_TEXT))
+  if option_id not in cert_files_with_options:
+    raise Exception('RunTimeError:Invalid input:{}. Please select valid '
+                    'certificate'.format(option_id))
+  return cert_files_with_options[option_id]
+
+
+def revokeCertificate():
+  """ Revokes a leaf certificate and updates the CRL chain file. """
+
+  # Retrives the all certificates from certs directory.
+  cert_name = getCertNameToBlacklist()
+  logging.info('Certificate selected to blacklist is:%s', cert_name)
+  print "select the certificate type:"
+  print "\n".join("[%s] %s" % (cert_id, cert_type)
+                  for (cert_id, cert_type) in sorted(CERT_TYPE.iteritems()))
+  cert_type = raw_input(CLI_PROMPT_TEXT)
+  if cert_type not in CERT_TYPE:
+    raise Exception("RunTimeError:Type of certificate to be blacklisted is incorrect."
+                    "Please select one of the values:%s" % CERT_TYPE.values())
+
+  revoke_cert_command = "cd {0} && ./revoke_and_generate_crl.sh " \
+                        "{1} {2}".format(getCertsDirectoryAbsolutePath(),
+                                         CERT_TYPE[cert_type],
+                                         cert_name)
+  command_exit_status = subprocess.call(revoke_cert_command, shell=True)
+  if command_exit_status:
+    raise Exception("RunTimeError:Certificate Revocation is failed:"
+                    "exit_code:{}".format(command_exit_status))
+  logging.info('%s is blacklisted successfully', cert_name)
+
+
+def updateCrlUrlAndRegenerateCerts():
+  """ Updates URI.0 field in the in openssl.cnf file with the CRL URL of this simple CRL server and
+  regenerates the certificates. """
+
+  # Update CRL URL in openssl.cnf.
+  crl_url = 'URI.0 = {}'.format(DEFAULT_CRL_URL)
+  update_crl_section_command = 'sed -i -e \'s~^URI.0\\s=\\s.*$~{}\'~1 ../../cert/openssl.cnf'.\
```

Comment:
Use a `r"raw string"` to avoid excess backslashes.

PEP8 says "Long lines can be broken over multiple lines by **wrapping expressions in parentheses**. These should be used **in preference to using a backslash for line continuation**."

Reply:
Changed   accordingly